### ISSUE TITLE
Firebase Core mis-match 0.6.0 to 1.7.0 vs. SDK 8.6.0

### ISSUE THREAD

**Issue by jbryanh** [FirebaseExtended/flutterfire#7179]
## Bug report

**Describe the bug**
On https://firebase.flutter.dev/docs/overview/ instructions are to set firebase_core to 0.7.0 and the quick download sets the SDK to 8.6.0, but the newest core version is 1.7.0 and according to https://pub.dev/packages/firebase_core/changelogm it bumps the firebase ios SDK to 8.7.0.   I haven't actually tried, but I would suppose 0.7.0 does not work with the 8.6.0 SDK, as 1.6.0 bumped to that.  I was set to 1.1 and 8.0.0 before needing to update today, and ran into issues without proper guidance on the web.

I now set dependencies to 1.6.0 and 8.6.0, which is present nowhere in documentation, but works.  1.7.0 and 8.7.0 fails as of today.  The 0.7.0 could be a typo that is meant to be 1.7.0...it's important to note that this fails.  I got an unresponsive error...could have been a blip, but to be safe I just bumped back to 1.6.0 and 8.6.0 and it worked.

Could you please update the proper guidance on flutterfire front page for the most updated, functioning combination of versions?  I don't need any support, you can close this as you wish, Just giving feedback and reference for future users.

### Steps to reproduce

2. '...'
4. See error or incorrect behavior

### Sample project

Providing a minimal example project which demonstrates the bug in isolation from your main App _greatly_ enhances the chance of a timely fix.
Please link to the public repository URL.

---

## Additional context

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
PASTE OUTPUT INSIDE HERE
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
PASTE OUTPUT INSIDE HERE
```

</details>

---

**Comment by jbryanh** [comment]
> @jbryanh Can you provide `flutter doctor -v`, `flutter pub deps -- --style=compact` and proper steps to replicate this issue ? Thanks.

Guessing that's just a bot reply because I didn't include those items.  This is a web site documentation issue.  I'm running fine on my system, my point is that the docs aren't updated.

**Comment by virakyuthpk** [comment]
Hello I'm look for a help on the same issue here,

Here is my response after `pos install`
```
Analyzing dependencies
firebase_core: Using Firebase SDK version '8.7.0' defined in 'firebase_core'
firebase_messaging: Using Firebase SDK version '8.7.0' defined in 'firebase_core'
[!] CocoaPods could not find compatible versions for pod "Firebase/CoreOnly":
  In Podfile:
    firebase_core (from `.symlinks/plugins/firebase_core/ios`) was resolved to 1.7.0, which depends on
      Firebase/CoreOnly (= 8.7.0)

None of your spec sources contain a spec satisfying the dependency: `Firebase/CoreOnly (= 8.7.0)`.

You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.
```

and here is my `Podfile`
```
# Uncomment this line to define a global platform for your project
platform :ios, '10.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end
end
```

and here is my `pubspec.yaml`
```
name: safety
description: A new Flutter project.

# The following line prevents the package from being accidentally published to
# pub.dev using `flutter pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number used as CFBundleVersion.
# Read more about iOS versioning at
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
version: 1.0.0+1

environment:
  sdk: ">=2.12.0 <3.0.0"

# Dependencies specify other packages that your package needs in order to work.
# To automatically upgrade your package dependencies to the latest versions
# consider running `flutter pub upgrade --major-versions`. Alternatively,
# dependencies can be manually updated by changing the version numbers below to
# the latest version available on pub.dev. To see which dependencies have newer
# versions available, run `flutter pub outdated`.
dependencies:
  flutter:
    sdk: flutter


  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.2
  flutter_local_notifications: ^9.0.0
  firebase_messaging: ^10.0.8
  firebase_core: ^1.7.0

dev_dependencies:
  flutter_test:
    sdk: flutter

  # The "flutter_lints" package below contains a set of recommended lints to
  # encourage good coding practices. The lint set provided by the package is
  # activated in the `analysis_options.yaml` file located at the root of your
  # package. See that file for information about deactivating specific lint
  # rules and activating additional ones.
  flutter_lints: ^1.0.0

# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

# The following section is specific to Flutter.
flutter:

  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true

  # To add assets to your application, add an assets section, like this:
  # assets:
  #   - images/a_dot_burr.jpeg
  #   - images/a_dot_ham.jpeg

  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.dev/assets-and-images/#resolution-aware.

  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages

  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  # fonts:
  #   - family: Schyler
  #     fonts:
  #       - asset: fonts/Schyler-Regular.ttf
  #       - asset: fonts/Schyler-Italic.ttf
  #         style: italic
  #   - family: Trajan Pro
  #     fonts:
  #       - asset: fonts/TrajanPro.ttf
  #       - asset: fonts/TrajanPro_Bold.ttf
  #         weight: 700
  #
  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages
```

and here is my `flutter doctor`
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.5.2, on macOS 11.5.2 20G95 darwin-arm, locale en-KH)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0-rc1)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 2020.3)
[✓] Connected device (2 available)

• No issues found!
```

**Comment by virakyuthpk** [comment]
> @virakyuthpk I'm using 1.6.0 with no problems if you want to just get up and running. [I download the SDK 8.6.0 to improve ios build times.](https://firebase.flutter.dev/docs/overview/#improve-ios-build-times)

Now I'm using 
```
flutter_local_notifications: ^9.0.0
firebase_messaging: ^10.0.8
firebase_core: ^1.6.0
```
but still not working and here is the output from command ``flutter build ios``
```
Building com.chipmong.safety for device (ios-release)...
Automatically signing iOS for device deployment using specified development team in Xcode project: 427PB6877P
Running pod install...                                           1,039ms
CocoaPods' output:
↳
      Preparing

    Analyzing dependencies

    Inspecting targets to integrate
      Using `ARCHS` setting to build architectures of target `Pods-Runner`: (``)

    Finding Podfile changes
      A firebase_core
      A firebase_messaging
      A flutter_local_notifications
      - Flutter
      - device_info_plus
      - images_picker
      - path_provider
      - sqflite
      - sqlite3_flutter_libs

    Fetching external sources
    -> Fetching podspec for `Flutter` from `Flutter`
    -> Fetching podspec for `device_info_plus` from `.symlinks/plugins/device_info_plus/ios`
    -> Fetching podspec for `firebase_core` from `.symlinks/plugins/firebase_core/ios`
    firebase_core: Using Firebase SDK version '8.7.0' defined in 'firebase_core'
    -> Fetching podspec for `firebase_messaging` from `.symlinks/plugins/firebase_messaging/ios`
    firebase_messaging: Using Firebase SDK version '8.7.0' defined in 'firebase_core'
    -> Fetching podspec for `flutter_local_notifications` from `.symlinks/plugins/flutter_local_notifications/ios`
    -> Fetching podspec for `images_picker` from `.symlinks/plugins/images_picker/ios`
    -> Fetching podspec for `path_provider` from `.symlinks/plugins/path_provider/ios`
    -> Fetching podspec for `sqflite` from `.symlinks/plugins/sqflite/ios`
    -> Fetching podspec for `sqlite3_flutter_libs` from `.symlinks/plugins/sqlite3_flutter_libs/ios`

    Resolving dependencies of `Podfile`
      CDN: trunk Relative path: CocoaPods-version.yml exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_d_c_2.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/d/c/2/sqlite3/3.36.0/sqlite3.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/d/c/2/sqlite3/3.36.0/sqlite3.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/d/c/2/sqlite3/3.36.0/sqlite3.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/d/c/2/sqlite3/3.36.0/sqlite3.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/d/c/2/sqlite3/3.36.0/sqlite3.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_0_3_5.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/3/5/Firebase/7.9.0/Firebase.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_f_4_e.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/f/4/e/FMDB/2.7.5/FMDB.podspec.json exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: all_pods_versions_0_a_a.txt exists! Returning local because checking is only performed in repo update
      CDN: trunk Relative path: Specs/0/a/a/ZLPhotoBrowser/4.1.9/ZLPhotoBrowser.podspec.json exists! Returning local because checking is only performed in repo update
    [!] CocoaPods could not find compatible versions for pod "Firebase/CoreOnly":
      In Podfile:
        firebase_core (from `.symlinks/plugins/firebase_core/ios`) was resolved to 1.7.0, which depends on
          Firebase/CoreOnly (= 8.7.0)

    None of your spec sources contain a spec satisfying the dependency: `Firebase/CoreOnly (= 8.7.0)`.

    You have either:
     * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
     * mistyped the name or version.
     * not added the source repo that hosts the Podspec to your Podfile.

    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:317:in `raise_error_unless_state'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:299:in `block in unwind_for_conflict'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:297:in `tap'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:297:in `unwind_for_conflict'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:257:in `process_topmost_state'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolution.rb:182:in `resolve'
    /Library/Ruby/Gems/2.6.0/gems/molinillo-0.8.0/lib/molinillo/resolver.rb:43:in `resolve'
    /Users/Sam/.gem/ruby/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/resolver.rb:94:in `resolve'
    /Users/Sam/.gem/ruby/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer/analyzer.rb:1078:in `block in resolve_dependencies'
    /Users/Sam/.gem/ruby/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/Sam/.gem/ruby/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer/analyzer.rb:1076:in `resolve_dependencies'
    /Users/Sam/.gem/ruby/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer/analyzer.rb:124:in `analyze'
    /Users/Sam/.gem/ruby/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:416:in `analyze'
    /Users/Sam/.gem/ruby/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:241:in `block in resolve_dependencies'
    /Users/Sam/.gem/ruby/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/user_interface.rb:64:in `section'
    /Users/Sam/.gem/ruby/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:240:in `resolve_dependencies'
    /Users/Sam/.gem/ruby/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/installer.rb:161:in `install!'
    /Users/Sam/.gem/ruby/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/command/install.rb:52:in `run'
    /Library/Ruby/Gems/2.6.0/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
    /Users/Sam/.gem/ruby/2.6.0/gems/cocoapods-1.11.2/lib/cocoapods/command.rb:52:in `run'
    /Users/Sam/.gem/ruby/2.6.0/gems/cocoapods-1.11.2/bin/pod:55:in `<top (required)>'
    /usr/local/bin/pod:23:in `load'
    /usr/local/bin/pod:23:in `<main>'

Error: CocoaPods's specs repository is too out-of-date to satisfy dependencies.
To update the CocoaPods specs, run:
  pod repo update



Error running pod install

```

**Comment by russellwheatley** [comment]
Hey @jbryanh, thanks for the feedback. We have updated our documentation so that there is no mention of the specific version of `firebase_core` to install.

Instead, we have advised users to update the dependency like so:

```bash
flutter pub add firebase_core
``` 
which will install the latest version. Thanks.
